tests/html_glimmer/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -258,10 +264,18 @@ exports[`element-node.hbs 1`] = `
   A long enough string to trigger a line break that would prevent wrapping more and more.
 </div>
 <div>
+  A line break with its own new lines
+  inside of it forming something of 
+  a paragraph.
```

Comment:
For this, we use the `fill` primitive which adds whitespace when the first word would hit the limit. https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L4413-L4452

Reply:
Hmm, so using fill would require `split` on the text characters because fill takes an array. One method is something like, `fill(n.chars.split(/\s+/).map(a => elem + " ")`, but that means that if a user has HTML which has double spacing in it, then we are going  to chew it away. Ex:

`<div>Hello   World</div>`
Becomes:
`<div>Hello World</div>`

I think we should do this. It's much better than preserving multiple spaces, because that gets ignored by HTML anyway, outside of `<pre>` tags.

Reply:
Someone posted a long comment as to why whitespace and html is going to be difficult. For example you can apply `white-space: pre` to arbitrary nodes in CSS which we have no way to know about when just looking at the template.

This means that the only want to guarantee correctness is not to change anything, which defeats the purpose. So we should use good judgement as to when we diverge and when we respect it and accept the fact that we can't get it right all the time.

Reply:
Also, if you know that you’re going to set `white-space: pre`, you can put a `<!-- prettier-ignore -->` before the element you apply it to.

Reply:
Alright, those are great points. I'm going to pretend `<pre>` and `white-space: pre` don't exist for the purpose of developing something that's more useful to the user anyway.